pkg/logqlmodel/stats/context.go

```diff
@@ -139,6 +139,9 @@ func (r *Result) ComputeSummary(execTime time.Duration) {
 			int64(float64(r.Summary.TotalLinesProcessed) /
 				execTime.Seconds())
 	}
+	if enqueueTime != 0 {
+		r.Summary.EnqueueTime = enqueueTime.Seconds()
```

Comment:
Instead of using `<time.Duration>.Seconds()`, we can just pass this as `int64(<time.Duration>)`, which is how `time.Duration` is defined (`type Duration int64`). This lets us store the enqueue time in a loss-less way.

Reply:
Good point 👍 
Addressed in commit: https://github.com/grafana/loki/pull/4949/commits/f63cc6ef6be9f00b77732606d8286fba621bb6fa